sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -75,18 +81,52 @@ case class CreateTableLikeCommand(
       throw new AnalysisException(
         s"Source table in CREATE TABLE LIKE does not exist: '$sourceTable'")
     }
-    if (catalog.isTemporaryTable(sourceTable)) {
-      throw new AnalysisException(
-        s"Source table in CREATE TABLE LIKE cannot be temporary: '$sourceTable'")
-    }
 
-    val tableToCreate = catalog.getTableMetadata(sourceTable).copy(
-      identifier = targetTable,
-      tableType = CatalogTableType.MANAGED,
-      createTime = System.currentTimeMillis,
-      lastAccessTime = -1).withNewStorage(locationUri = None)
+    val sourceTableDesc = catalog.getTableMetadata(sourceTable)
+    val sourceStorageFormat = sourceTableDesc.storage
 
-    catalog.createTable(tableToCreate, ifNotExists)
+    sourceTableDesc.tableType match {
+      case CatalogTableType.MANAGED | CatalogTableType.EXTERNAL | CatalogTableType.VIEW => // OK
+      case o => throw new AnalysisException(
+        s"CREATE TABLE LIKE is not allowed when the source table is ${o.name}")
+    }
+
+    // For EXTERNAL_TABLE, the table properties has a particular field. To change it
+    // to a MANAGED_TABLE, we need to remove it; Otherwise, it will be EXTERNAL_TABLE,
+    // even if we set the tableType to MANAGED
+    // (metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L1095-L1105)
+    // Table comment is stored as a table property. To clean it, we also should remove them.
+    val newTableProp = sourceTableDesc.properties.filterKeys(_ != "EXTERNAL")
```

Comment:
yea, but in `HiveClientImply` instead of `HiveExternalCatalog`, to match the place where we put it.